ui/src/pages/clearing/AllocationModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from 'react';
+import { useLedger } from '@daml/react';
+import { CreateEvent } from '@daml/ledger';
+import {
+  AssetDeposit,
+  AssetDeposit_SetObservers,
+  AssetDeposit_Split,
+} from '@daml.js/da-marketplace/lib/DA/Finance/Asset';
+import { Button, Form, Header, Modal } from 'semantic-ui-react';
+
+type MarginCallProps = {
+  deposit: CreateEvent<AssetDeposit> | undefined;
+  title: string;
+  open: boolean;
+  onClose: (open: boolean) => void;
+  observers: AssetDeposit_SetObservers;
+};
+
+const AllocationModal: React.FC<MarginCallProps> = ({
+  deposit,
+  title,
+  open,
+  onClose,
+  observers,
+}) => {
+  const ledger = useLedger();
+  const [allocation, setAllocation] = useState<number>(0.0);
+
+  useEffect(() => {
+    if (!!deposit) setAllocation(parseFloat(deposit.payload.asset.quantity));
+  }, [open, deposit]);
+
+  if (!deposit) return <></>;
```

Comment:
better to `return null` so that you don't render anything from this component in this case

Reply:
done